zero/utils/type_util.py

```diff
@@ -209,4 +209,14 @@ def is_allowed_type(typ: Type):
         if issubclass(typ, mtype):
             return True
 
+    if _is_pydantic_model_type(typ):
+        return True
+
     return False
+
+
+def _is_pydantic_model_type(typ: Type) -> bool:
+    return any(
+        base.__module__.startswith("pydantic") and base.__name__ == "BaseModel"
```

Comment:
Pydantic also has a `GenericModel` do we need to consider that?

Reply:
Good point indeed! I wanted to boot zero up only when rpc functions are properly formatted, to eliminate parsing/marshalling errors. If we totally remove `is_allowed_type` then there's one more layer of debugging people might need to do when zero server is running properly. We can probably add this in the Encoder Protocol, wdyt? If someone wants to use a new Encoder, they can extend what we offer, but then they are responsible for supporting the types, otherwise we affect default people with the burden. wdyt?